single_include/glob/glob.hpp

```diff
@@ -181,7 +181,12 @@ bool has_magic(const std::string &pathname) {
 }
 
 static inline 
-bool is_hidden(const std::string &pathname) { return pathname[0] == '.'; }
+bool is_hidden(const std::string &pathname) {
+  // path is hidden if it starts with dot and is not parent dir nor current dir
+  if (pathname.size() >= 2)
+    return pathname.at(0) == '.' && pathname.at(1) != '.' && pathname.at(1) != '/';
+  return pathname[0] == '.';
+}
```

Comment:
Hi,
indeed I agree that my pretty basic implementation would fail. However to be fair, the actual implementation would also fail, returning `false` for any path that starts with `.` such as for example `../foo` or `./bar`, leading to the path not being searched at all using the `glob/rglob` functions. I wanted a very simple and nonintrusive implementation that fit my needs of globing parent paths but I'm very open to alternatives.

I personally think that the function should look for any directory that is hidden within the given path and not only the first character. If any of those directories begins with dot and followed by something else than dot or slash, then the function should consider this path as hidden.

Also but unrelated, unless there is verification somewhere else, `is_hidden` will crash if given an empty string as parameter (line 188 accesses index 0 without checking if pathname is empty).

BTW, thanks for the library, very useful and simple to integrate!

Reply:
Agreed. I'll rework my PR to support those.

Reply:
Great, thanks!